src/structures/game_session.ts

```diff
@@ -1181,6 +1193,55 @@ export default class GameSession {
         this.addBookmarkedSong(interaction.member?.id, song);
     }
 
+    /**
+     * If the game changes its premium state, update filtered songs
+     * @param premiumJoined - true if a premium member joined VC
+     */
+    async updatePremiumStatus(premiumJoined: boolean): Promise<void> {
+        if (premiumJoined) {
+            this.premiumGame = true;
```

Comment:
Don't we still need to reload songs if game wasn't premium before this?

Reply:
Somewhat improved this function